### PR TITLE
Add hackstead/new and hackstead/remove routes + tests

### PR DESCRIPTION
**Pull request by cedric-h**

### CODE REVIEW

File: src/hackstead/mod.rs [hunk 1]
```diff
@@ -1,27 +1,55 @@
...
+use crate::data;
+use actix_web::{get, post, web, HttpResponse};
+use log::*;
 use hcor::{Hackstead, ServiceError, UserContact};
 
+#[cfg(test)]
+mod test;
+
 #[get("/hackstead/")]
 pub async fn get_hackstead(form: web::Json<UserContact>) -> Result<HttpResponse, ServiceError> {
```
Comment (Muirrum) [inline src/hackstead/mod.rs:10]: This should take in a user request, no?

File: src/hackstead/mod.rs [hunk 2]
```diff
@@ -1,27 +1,55 @@
...
-    log::debug!("servicing get_hackstead request");
```
Comment (Muirrum) [inline src/hackstead/mod.rs:6]: This makes sense to keep in as a debug log

File: src/hackstead/mod.rs [hunk 3]
```diff
@@ -1,27 +1,55 @@
...
-    log::debug!("hackstead: {:?}", stead);
+    trace!("got hackstead: {:#?}", stead);
 
     Ok(HttpResponse::Ok().json(stead))
 }
 
-#[cfg(test)]
-mod test;
+#[post("/hackstead/new")]
+pub async fn new_hackstead(form: web::Json<UserContact>) -> Result<HttpResponse, ServiceError> {
```
Comment (Muirrum) [inline src/hackstead/mod.rs:28]: Again this should take in a UserRequest

**Review by Muirrum — CHANGES_REQUESTED** [review]

File: src/hackstead/mod.rs (hunk 1 above)
Comment (cedric-h) [reply src/hackstead/mod.rs:10]: I deleted `UserRequest`s a long time ago, felt redundant when we have `UserContact`. Is there some nuance there that I'm not appreciating?

File: src/hackstead/mod.rs (hunk 2 above)
Comment (cedric-h) [reply src/hackstead/mod.rs:6]: yeah, you're probably right about that. I removed it remembering incorrectly that async-web would do a better job of logging when the route was called automatically.

File: src/hackstead/mod.rs (hunk 1 above)
Comment (Muirrum) [reply src/hackstead/mod.rs:10]: IMO `UserRequest` is less data for the client to have to send in, as opposed to the contact data. I see where you're coming from though

Comment (cedric-h) [reply src/hackstead/mod.rs:10]: I mean with `UserRequest` they need to specify either a slack id or an id that we coined
with `UserContact` it's possible to specify both, whereupon we arbitrarily select one or the other (probably the one we have a mongodb index on?)
so I think we ought to just stick with `UserContact` so we never even have to define/expose a `UserRequest`.

**Review by Muirrum — APPROVED** [review]

**Pull request merged**